Data-Structure/Algorithm-Design-Part-1/DictionaryUsingBST/BSTNode.java

```diff
@@ -0,0 +1,55 @@
+package DictionaryUsingBST;
+
+/**
+ * @author Niraj Gupta
+ * Class to create a node which is inserted in the Tree. Each node has key and value and reference to their 
+ * left and right child node.
+ * 
+ * @param <K> : The Data Type assigned to the key of the node.
+ * @param <V> : The Data Type assigned to the value of the node.
+ */
+public class BSTNode<K extends Comparable<K>, V>{
+	protected K key;
+	protected V value;

```

Comment:
if  needed we can always add inline comments for explaining the var use like we add method specs fr each method but avoid  inclusion at cls level. Reason of this is if be the case and we add specs fr all vars and methods at cls level it would make no sense to add it again at granular level then. Also, it makes the class spec hard to read

Reply:
Ok will focus on inline comment...